### PR TITLE
add test cases for clean_banner

### PR DESCRIPTION
**Pull request by justinbastress**

### CODE REVIEW

File: ztag/test/encoding_test.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                v = k
+            temp = v
+            for ip3 in (121, 122):
+                for ip4 in range(0, 256):
+                    for worker in range(1, 10):
+                        kk = k.format(worker=worker, ip3=ip3, ip4=ip4)
+                        vv = v.format(worker=worker, ip3=ip3, ip4=ip4)
+                        cases[kk] = vv
+
+        for input, expected in cases.items():
```
Comment (parkanzky) [inline ztag/test/encoding_test.py]: This overwrites built-in `input`.

Comment (justinbastress) [reply ztag/test/encoding_test.py]: Good old python -- making the assumption that no one would use obscure variable names like `input`, `file` or `dir`.

File: ztag/test/encoding_test.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            # SFJ: IP appears twice on different lines
+            "Got connection from 198.108.66.{ip4}\nrejecting 198.108.66.{ip4}.": "Got connection from CLIENT_IP\nrejecting CLIENT_IP.",
+            # SFJ: Host appears twice on different lines
+            "Got connection from worker-{worker}.sfj.corp.censys.io\nrejecting worker-{worker}.sfj.corp.censys.io.": "Got connection from CLIENT_HOSTNAME\nrejecting CLIENT_HOSTNAME."
+
+        }
+        cases = {}
+        for k, v in cases_format.items():
+            if not v:
+                cases_format[k] = k
```
Comment (parkanzky) [inline ztag/test/encoding_test.py]: Is this assignment superfluous? I don't see where `cases_format[k]` is ever used again.

File: ztag/test/encoding_test.py (hunk 1 above)
Comment (andrewsardone) [reply ztag/test/encoding_test.py]: > Good old python -- making the assumption that no one would use obscure variable names like `input`, `file` or `dir`.

The more you know!

**Pull request merged**